arch/risc-v/include/syscall.h

```diff
@@ -207,7 +162,15 @@ static inline uintptr_t sys_call0(unsigned int nbr)
 
   asm volatile
     (
+#if defined (CONFIG_ARCH_USE_S_MODE) && defined (__KERNEL__)
+     " addi sp, sp, -16\n"                  /* Make room */
+     " sd   ra, 0(sp)\n"                    /* Save ra */
```

Comment:
Do we need `sw` vs `sd` selection here

Reply:
Yes, this works for 64-bit only, oversight by me.

The handy macros REGLOAD/REGSTORE are not usable from inline assembly if I'm right ?

Reply:
I think we do not. I need to check. Some time ago I had a plan to move those macro to a header file and add some `#s` wrapper for inline assembly, but didn't find time for that

Reply:
I can try using a stringify macro like above, it will save a lot of copypaste

Reply:
Or we can leave it as is and do a refactoring PR right after this one

Reply:
I think we need to move such things to `arch/risc-v/src/common/riscv_internal.h` and have something like
```
#ifndef __ASSEMBLY__
#. define _STR(s)        #s
#else
#. define _STR(s)        s
#endif

#define _XSTR(s)       _STR(s)

#if defined(CONFIG_ARCH_QPFPU)
#  define FLOAD     _XSTR(flq)
#  define FSTORE    _XSTR(fsq)
#elif defined(CONFIG_ARCH_DPFPU)
#  define FLOAD     _XSTR(fld)
#  define FSTORE    _XSTR(fsd)
#else
#  define FLOAD     _XSTR(flw)
#  define FSTORE    _XSTR(fsw)
#endif

#ifdef CONFIG_ARCH_RV32
#  define REGLOAD   _XSTR(lw)
#  define REGSTORE  _XSTR(sw)
#else
#  define REGLOAD   _XSTR(ld)
#  define REGSTORE  _XSTR(sd)
#endif
```

Reply:
but that is a more refactoring and is not related to this PR

Reply:
To me it is fine either way.

Reply:
Yes that would be the way to do it.

Reply:
Ok. I will submit the change

Reply:
> Ok. I will submit the change

Nice, thanks for the help

Reply:
https://github.com/apache/incubator-nuttx/pull/5917 -- please take a look